### PR TITLE
refactor: Remove unsafe access to ZMUserSession.shared()! - WPB-5331

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-5331" title="WPB-5331" target="_blank">BugWPB-5331</a>  Remove unsafe access to ZMUserSession.shared()!
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we remove unsafe access to ZMUserSession.shared()! by using methods like guard let to safely unwrap the optional. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/Model/FeatureConfig/AppLock/AppLockController.swift [hunk 1]
```diff
@@ -119,7 +119,7 @@ public final class AppLockController: AppLockType {
     public init(
         userId: UUID,
         selfUser: ZMUser,
-        legacyConfig: LegacyConfig?
+        legacyConfig: LegacyConfig? = nil
```
Comment (markusfassbender) [inline wire-ios-data-model/Source/Model/FeatureConfig/AppLock/AppLockController.swift]: I have removed this in my PR, maybe there was merge conflict?

File: wire-ios/Wire-iOS/Sources/AppRootRouter.swift [hunk 2]
```diff
@@ -408,7 +408,10 @@ extension AppRootRouter {
         userSession: UserSession,
         isComingFromRegistration: Bool
     ) -> AuthenticatedRouter? {
-        let isTeamMember: Bool
+        let needToShowDataUsagePermissionDialog = appStateCalculator.wasUnauthenticated && !SelfUser.current.isTeamMember
+        guard let userSession = ZMUserSession.shared() else { return  nil }
+
+      let isTeamMember: Bool
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/AppRootRouter.swift]: `isTeamMember` is duplicated now and also `!SelfUser.current.isTeamMember` doesn't exist anymore, it's now `SelfUser.provider?.providedSelfUser`. I guess the logic here must be updated to work, maybe it was a merge conflict?

File: wire-ios/Wire-iOS/Sources/Helpers/syncengine/ZiphySession.swift [hunk 3]
```diff
@@ -30,19 +37,31 @@ extension ZMUserSession: ZiphyURLRequester {
...
 
-    private func doRequest(withPath path: String, method: ZMTransportRequestMethod, type: ProxiedRequestType, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> ProxyRequest {
-        return proxiedRequest(path: path, method: method, type: type, callback: completionHandler)
+    private func doRequest(
+        withPath path: String,
+        method: ZMTransportRequestMethod,
+        type: ProxiedRequestType,
+        completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void
+    ) -> ProxyRequest {
+        return userSession.proxiedRequest(
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/Helpers/syncengine/ZiphySession.swift]:
```suggestion
        userSession.proxiedRequest(
```
Return can be omitted here, not sure if we want it?

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DatabaseStatisticsController.swift [hunk 4]
```diff
@@ -80,8 +80,8 @@ final class DatabaseStatisticsController: UIViewController {
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
-        let session = ZMUserSession.shared()
-        let syncMoc = session!.managedObjectContext.zm_sync!
+        guard let session = ZMUserSession.shared() else { return }
```
Comment (markusfassbender) [inline wire-ios/Wire-iOS/Sources/UserInterface/Settings/DatabaseStatisticsController.swift:83]: Do we want to add something like `assertionFailure` here to crash in debug builds?

**Review by markusfassbender — CHANGES_REQUESTED** [review]

File: wire-ios/Wire-iOS/Sources/AppRootRouter.swift (hunk 2 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/AppRootRouter.swift]: Fixed it!

File: wire-ios-data-model/Source/Model/FeatureConfig/AppLock/AppLockController.swift (hunk 1 above)
Comment (agisilaos) [reply wire-ios-data-model/Source/Model/FeatureConfig/AppLock/AppLockController.swift]: Fixed it!

File: wire-ios/Wire-iOS/Sources/UserInterface/Settings/DatabaseStatisticsController.swift (hunk 4 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/UserInterface/Settings/DatabaseStatisticsController.swift:83]: I don't think it would be necessary in this case.

**Review by markusfassbender — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/AppRootRouter.swift [hunk 5]
```diff
@@ -408,7 +408,9 @@ extension AppRootRouter {
         userSession: UserSession,
         isComingFromRegistration: Bool
     ) -> AuthenticatedRouter? {
-        let isTeamMember: Bool
+        guard let userSession = ZMUserSession.shared() else { return  nil }
+
+      let isTeamMember: Bool
```
Comment (David-Henner) [inline wire-ios/Wire-iOS/Sources/AppRootRouter.swift]: indentation looks off

**Review by David-Henner — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/AppRootRouter.swift (hunk 5 above)
Comment (agisilaos) [reply wire-ios/Wire-iOS/Sources/AppRootRouter.swift]: Done!

**Pull request merged**